core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -80,6 +81,28 @@ static int ti_sci_setup_xfer(uint16_t msg_type, uint32_t msg_flags,
 	return 0;
 }
 
+/**
+ * ti_sci_send_request() - Send request to mailbox channel
+ *
+ * @xfer:	Transfer to initiate and wait for response
+ *
+ * Return: 0 if all goes well, else appropriate error message
+ */
+static inline int ti_sci_send_request(struct ti_sci_xfer *xfer)
+{
+	struct k3_sec_proxy_msg *msg = &xfer->tx_message;
+	struct ti_sci_msg_hdr *hdr = NULL;
+
+	hdr = (struct ti_sci_msg_hdr *)msg->buf;
+	hdr->seq = ++message_sequence;
```

Comment:
I haven't changed the following @etienne-lms , if you could let me know the reasoning as to why we need to split it into two and let the compiler do the optimisations..

Reply:
Easier to read. Prefix increment is a bit rare, so having it here makes one stop for just an instant to parse out "oh yeah, that means update *then* assign", vs just updating then assigning written out in the code. Code is for humans, the compiler will produce the same assembly in both cases, so why be fancy when you can be explicit :)

Reply:
Ah okay, makes sense. Would do the need